docs: Add info on duplicate column handling for GraphQL unnesting (#305)
spiceai/docs: spiceaidocs/docs/data-connectors/graphql.md

+------------------+--------+---------------+
```

#### Unnesting Duplicate Columns

> By default, the Spice Runtime will error when a duplicate column is detected during unnesting.

Reviewer: Can you provide an example of the runtime error?

Reviewer 2: Yes, providing example output would also be helpful.

For example, this example `spicepod.yml` query would fail due to `name` fields:
```yaml
from: graphql:https://localhost
name: stargazers
params:
  unnest_depth: 2
  json_path: data.users
  query: |
    query {
      users {
        name
        metadata {
          name: email
        }
      }
    }
```

> To avoid this error, you should use [use aliases in your query](https://www.apollographql.com/docs/kotlin/advanced/using-aliases/) where possible. In the example above, we used an alias to induce the duplicate error in our query with `name: email`.

Reviewer: ```suggestion
Avoid this error by [using aliases in the query](https://www.apollographql.com/docs/kotlin/advanced/using-aliases/) where possible. In the example above, a duplicate error in our query was induced with `name: email`.
```

Reviewer: Even with these edits this part:
we used an alias to induce the duplicate error in our query with `name: email`.

Is not clear.  There is no concrete example where the naming is the same?  I think inducing it is a little confusing.  Can you reword or provide a different example?

Note the edits are to have a more objective voice.  Try to use this in all documentation.

> The example below is more complex where a query returns a person, their name, and their starships' name. We've used an alias to rename `starship.name` to `starshipName` when unnesting, to avoid the duplicate column error:

Reviewer: ```suggestion
The example below is more complex where a query returns a person, their name, and their starships' name. An alias is used to rename `starship.name` to `starshipName` when unnesting, to avoid the duplicate column error:
```
